### PR TITLE
First chapter

### PR DESCRIPTION
**Pull request by XtraKrispi**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -473,7 +474,8 @@ Implement a function that returns the last digit of a given number.
   whether it works for you!
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = (abs n) `mod` 10
```
Comment (chshersh) [inline src/Chapter1.hs]: Nice usage of infix function call with backticks! We haven't explained this syntax to keep the tutorial short, but for functions like `div` and `mod` I usually prefer such notation for cleaner code :slightly_smiling_face: 

Also, `()` are redundant here, GHC can figure it out, but there's no harm in leaving them either.
```suggestion
lastDigit n = abs n `mod` 10
```

File: src/Chapter1.hs [hunk 2]
```diff
@@ -551,7 +559,13 @@ True
...
 False
 -}
-isVowel c = error "isVowel: not implemented!"
+isVowel :: Char -> Bool
+isVowel c | c == 'a' = True
+          | c == 'e' = True
+          | c == 'i' = True
+          | c == 'o' = True
+          | c == 'u' = True
+          | otherwise = False
```
Comment (chshersh) [inline src/Chapter1.hs]: Usually, when using guards, people write them on the next line with fixed indentation. So you don't need to change indent if you decide to rename the function in the future.

Of course, there's nothing wrong in your implementation, I'm just sharing some tips :slightly_smiling_face:

File: src/Chapter1.hs [hunk 3]
```diff
@@ -614,8 +628,13 @@ Implement a function that returns the sum of the last two digits of a number.
...
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = 
+  let lastTwo = (abs n) `mod` 100
+      lastOneInLastTwo = lastTwo `mod` 10
+      tensDigit = lastTwo `div` 10
+  in tensDigit + lastOneInLastTwo
```
Comment (chshersh) [inline src/Chapter1.hs]: Wow, that's a very clean solution!

Also, you might be interested in learning about the `divMod` function that returns a pair of `div` and `mod` results at the same time.

File: src/Chapter1.hs [hunk 4]
```diff
@@ -635,8 +654,9 @@ Implement a function that returns the first digit of a given number.
 You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n | (abs n) < 10 = abs n
+             | otherwise = firstDigit ((abs n) `div` 10)
```
Comment (chshersh) [inline src/Chapter1.hs]: Nice implementation! If you want go one step further, you can even create a helper recursive function and pass the result of `abs` only once. But that's just a minor improvement :slightly_smiling_face:

**Review by chshersh — APPROVED** [review]
Nice solutions! And kudos for solving the advanced task :star2: 

I left some comments, but they are just minor suggestions. Your solutions are great :+1:

File: src/Chapter1.hs (hunk 3 above)
Comment (XtraKrispi) [reply src/Chapter1.hs]: Awesome, I didn't realize `divMod` was a thing, I'll definitely rewrite this to use it!

File: src/Chapter1.hs [hunk 5]
```diff
@@ -631,8 +633,7 @@ specifying complex expressions.
 sumLast2 :: Int -> Int
 sumLast2 n = 
   let lastTwo = (abs n) `mod` 100
-      lastOneInLastTwo = lastTwo `mod` 10
-      tensDigit = lastTwo `div` 10
+      (tensDigit, lastOneInLastTwo) = lastTwo `divMod` 10
```
Comment (chshersh) [inline src/Chapter1.hs:636]: Wow, nice usage of pattern-matching together with `divMod`!

File: src/Chapter1.hs [hunk 6]
```diff
@@ -655,9 +656,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 firstDigit :: Int -> Int
-firstDigit n | (abs n) < 10 = abs n
-             | otherwise = firstDigit ((abs n) `div` 10)
-
+firstDigit n = go (abs n)
+  where go n' 
+          | n' < 10 = n'
+          | otherwise = go (n' `div` 10)
```
Comment (chshersh) [inline src/Chapter1.hs:662]: Nice and clean solution!

**Review by chshersh — APPROVED** [review]
Congrats on finishing Chapter One!
I think this PR is ready for merge :slightly_smiling_face:

**Pull request merged**